pkg/kubectl/cmd/create_service.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+// NewCmdCreateService is a macro command to create a new namespace
+func NewCmdCreateService(f *cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "service",
+		Short: "Create a service using specified subcommand.",
+		Long:  "Create a service using specified subcommand.",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.Help()
+		},
+	}
+	cmd.AddCommand(NewCmdCreateServiceClusterIP(f, cmdOut))
+	cmd.AddCommand(NewCmdCreateServiceNodePort(f, cmdOut))
+	cmd.AddCommand(NewCmdCreateServiceLoadBalancer(f, cmdOut))
+
+	return cmd
+}
+
+const (
+	serviceClusterIPLong = `
+Create a clusterIP service with the specified name.`
+
+	serviceClusterIPExample = `  # Create a new clusterIP service named my-cs
+  kubectl create service clusterip my-cs --tcp=5678:8080
+
+  # Create a new clusterIP service named my-cs (in headless mode)
+  kubectl create service clusterip my-cs --clusterip="None"`
+)
+
+func addPortFlags(cmd *cobra.Command) {
```

Comment:
> Because ports is a list, and this then will only allow one set which seems limiting. 
> If we do it this way you can pass in multiple port mappings.

Makes sense, works for me.